hooks/useWeatherData.ts

```diff
@@ -222,24 +211,55 @@ export const useWeatherData = (city: string, today: boolean) => {
     }
   };
 
+  const mainTemp = today ? mainTempToday : mainTempTomorrow;
+  const mainWeather = today
+    ? weatherData?.weather[0].description
+    : chooseMainWeatherforTomorrow();
+
+  const getForecast = () => {
+    let feelsLike: number | null = null;
+    let humidity: number | null = null;
+    let visibility: number | null = null;
+    let windSpeed: number | null = null;
+    let pressure: number | null = null;
+
+    if (today) {
+      const forecast = getForecastForTomorrow();
+      feelsLike = forecast.maxFeelsLike;
+      humidity = forecast.avgHumidity;
+      visibility = forecast.avgVisibility;
+      windSpeed = forecast.avgWindSpeed;
+      pressure = forecast.avgPressure;
+    } else {
+      feelsLike = weatherData?.main.feels_like ?? null;
+      humidity = weatherData?.main.humidity ?? null;
+      visibility = weatherData?.visibility ?? null;
+      windSpeed = weatherData?.wind.speed ?? null;
+      pressure = weatherData?.main.pressure ?? null;
+    }
+
+    return { feelsLike, humidity, visibility, windSpeed, pressure };
+  };
+
+  const weatherBackground = changeBackgroundImageDependsOnWeather();
+  const feelsLikeDescription = getfeelsLikeDescription();
+
   return {
     weatherData,
     forecastData,
-    loading,
-    error,
-    date,
-    todayMidnight,
-    tomorrowMidnight,
-    getHourlyForecastForNext24Hours,
-    getMinMaxTemp,
-    weatherDataForTomorrow,
-    chooseMainWeatherforTomorrow,
-    getForecastForTomorrow,
-    mainTempToday,
-    mainTempTomorrow,
+    weatherLoading,
+    weatherError,
```

Comment:
To, że ten hook robi dwa requesty (forecast i weather) to są bebechy, które raczej z zewnątrz nie powinny nas interesować. Wydaje się że moglibyśmy zwracać spokojnie po jednym loading i error stacie